CMakeBuildOptions.cmake

```diff
@@ -0,0 +1,86 @@
+# Put all top-level options build options into one place
```

Comment:
Should this comment go on to say that dependent and lower level options should not be entered in this file?

Reply:
That's not an absolute truth, there may be a good reason to put one here. Depends on the use case.

Reply:
ok